VogtObserver/Program.cs

```diff
@@ -8,33 +8,26 @@ class Program
         static void Main(string[] args)
         {
             RandomNumberGenerator generator = new RandomNumberGenerator();
-            Date dateTime = new Date();
-            Sum sum = new Sum();
+            Date dateTime = new Date(generator);
+            Sum sum = new Sum(generator);
```

Comment:
Here is where you'd want to do generator.AddObserver(dateTime);

Rather than passing the generator into the constructor.

Reply:
So you may ask.. what about dependency injection?

You don't always want to inject the dependency. In a lot of cases you do. But in this case, the **observer** pattern is abstracting the observer from the observable. They dont need to know/care about each other. Therefor,e this higher level function is "wiring" up those dependencies. This function assigns the observer to the observable.